framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -187,7 +189,55 @@ public int runTests() throws Exception {
       stopwatch.reset().start();
 
       LOG.info("> RUNNING TESTS (ITERATION " + i + ")..");
-      Collections.shuffle(tests, new Random());
+      String queryLogFilename = drillOutputDir + "/queryLog.log";
+      File queryFile = new File(queryLogFilename);
+      List<String> skippedTests = Lists.newArrayList();
+      if ((cmdParam.repeatRun == true) && (queryFile.exists() && (queryFile.length() > 0))) {
+        // generate a new list of tests in the requested order
+        try (BufferedReader reader = new BufferedReader(new FileReader(queryFile))) {
+          List<DrillTest> new_tests = Lists.newArrayList();
+          String line = "";
+          boolean found = false;
+          while ((line = reader.readLine()) != null) {
+            for (DrillTest test : tests) {
+              if (test.getInputFile().equals(line)) {
+                // verify that requested test is in the list of tests
+                boolean result = tests.remove(test);
+                if (result) {
+                  new_tests.add(test);
+                }
+                found = true;
+                break;
+              }
+            }
+            if (!found) {
+	      skippedTests.add(line);
+            }
+            found = false;
+          }
+          reader.close();
+          // throw away the original list of tests and use the new list
+          tests.clear();
+          tests = new_tests;
+        } catch (Exception e) {
+          LOG.error("\nCould not open queryFile containing tests to run");
+          LOG.error(e.getMessage());
+          System.exit(-1);
+        }
+      } else {
+        // shuffle the tests
+        Collections.shuffle(tests, new Random());
+        if (cmdParam.repeatRun == true) {
+          // record the tests and the order in which they are run
+          BufferedWriter writer = new BufferedWriter(new FileWriter(queryFile));
+          if (writer != null) {
+            for (DrillTest test : tests) {
+              writer.write (test.getInputFile() + "\n");
+            }
+            writer.close();
+          }
+        }
+      }
```

Comment:
Can you see if you can fit your code into the following logic structure:

if (cmdParam.repeatRun) {
  if (queryFile.length() > 0) {
    ...
  } else {
    ...
  }
}
original code...